### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by NoraLoose**
This is a WIP PR for more unit tests. We want to test all classes separately:

- [ ] `Case`
- [ ] `Component`
- [ ] `BaseModel`
- [ ] `AdditionalCode`
- [ ] `InputDataset`
- [ ] `Discretization`
- [ ] ...

### CODE REVIEW

File: cstar/tests/test_case.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+    @pytest.fixture
+    def case_with_mock_component(self):
+        """Fixture to create a Case object."""
+        mock_component = Mock()
+        mock_component.__class__.__name__ = "MockComponent"
+        mock_component.__repr__ = lambda self: "MockComponent()"
+
+        return Case(
+            components=[mock_component],
```
Comment (NoraLoose) [inline cstar/tests/test_case.py:145]: Once we have fixtures for all the separate objects (including Components), we can replace the mock component here by a real one.

**Review by TomNicholas — APPROVED** [review]

**Comment by NoraLoose** [comment]
Did you mean to merge this, and closed instead @dafyddstephenson?

**Comment by dafyddstephenson** [comment]
No, this should have been closed a while ago. The changes are now largely spread amongst other PRs targeting unit tests on individual modules.

**Comment by dafyddstephenson** [comment]
I'm now questioning why I didn't close it earlier and I think it's because there is still work here that hasn't been incorporated - in higher level classes like `Component` and `Case`. I can re-open and mark as a draft or keep closed and return to these changes when they are needed. I didn't want to start covering the higher level stuff until the foundational stuff was done.

**Comment by NoraLoose** [comment]
Makes sense. Just want to make sure the tests don't fall between the cracks.

**Comment by dafyddstephenson** [comment]
I think the only stuff missing was on `Case`, looking at it. I'll open a WIP PR today for `Case` unit tests using your commits as a start point.

**Pull request closed without merging**